### PR TITLE
Add Roslyn's new dev15.3.x branch to CI

### PR DESCRIPTION
**Pull request by shyamnamboodiripad**

### CODE REVIEW

**Comment by shyamnamboodiripad** [comment]
@jasonmalinowski

File: jobs/data/repolist.txt [hunk 1]
```diff
@@ -72,6 +72,8 @@ dotnet/roslyn branch=dev15.3-preview1 server=dotnet-ci
 dotnet/roslyn branch=dev15.3-preview1 subFolder=perf definitionScript=perf.groovy server=dotnet-ci2
 dotnet/roslyn branch=dev15.3-preview2 server=dotnet-ci
 dotnet/roslyn branch=dev15.3-preview2 subFolder=perf definitionScript=perf.groovy server=dotnet-ci2
```
Comment (shyamnamboodiripad) [inline jobs/data/repolist.txt]: @jasonmalinowski Can some of the above branches be removed now?

File: jobs/data/repolist.txt [hunk 2]
```diff
@@ -72,6 +72,8 @@ dotnet/roslyn branch=dev15.3-preview1 server=dotnet-ci
...
+dotnet/roslyn branch=dev15.3.x server=dotnet-ci
+dotnet/roslyn branch=dev15.3.x subFolder=perf definitionScript=perf.groovy server=dotnet-ci2
 dotnet/roslyn branch=dev15.6 server=dotnet-ci
 dotnet/roslyn branch=dev16 server=dotnet-ci
```
Comment (shyamnamboodiripad) [inline jobs/data/repolist.txt]: I am guessing it is by design that I am not seeing testimpact / LUT branches in this file. But is there some other file I need to modify to enable CI for the dev15.3.x LUT branch? @jasonmalinowski @genlu

**Comment by shyamnamboodiripad** [comment]
@dotnet/roslyn-infrastructure

File: jobs/data/repolist.txt (hunk 1 above)
Comment (jasonmalinowski) [inline jobs/data/repolist.txt]: @shyamnamboodiripad Yes, we can delete everything that's dev15.* that's not dev15.0.x and dev15.2.x.

**Pull request merged**